inst/scripts/SEL3D.R

```diff
@@ -0,0 +1,251 @@
+    RCSBid <- gsub("_", ".", blastOutput$pdb.id[entry], ignore.case = TRUE)
+    entryXMLTable <- XML::xmlParse(RCurl::getURL(paste
+                                     ("http://www.rcsb.org/pdb/rest/describeMol?structureId=",
+                                       RCSBid, sep= "")))
+
+    #get taxonomy
+
+    #We don't want to drop all the non-homo sapien matches because of
+    #the edge case where a match could have 100% amino acid ID for a
+    #non homo sapien which we want to include in the table.
+    #So we need to get all tax information as well as the protein
+    #information, without dropping anything solely based on tax
+
+    tax <- XML::getNodeSet(entryXMLTable, '//Taxonomy/@name')[[1]][[1]]
+    info <- XML::getNodeSet(entryXMLTable, '//polymerDescription/@description')[[1]][[1]]
+
+    if (length(tax) > 0) {
+      blastOutput$tax[entry] <- tax
+    } else {
+      blastOutput$tax[entry] <- NA
+    }
+
+    if(length(info) > 0) {
+      blastOutput$info[entry] <- info
+    } else {
+      blastOutput$info[entry] <- NA
+    }
+
+
+    if(!intersectingPositions(finalOutputRange, blastOutput$q.start[entry], blastOutput$q.end[entry])){
+      # Finally, we then find the best match that is either Homo sapiens or 100% ID.
+      if((!is.na(blastOutput$tax[entry]) && blastOutput$tax[entry] == "Homo sapiens")
+         || blastOutput$identity[entry] == 100){
+        finalOutput <- rbind(finalOutput, blastOutput[entry,])
+        finalOutputRange[[finalOutputRangePointer]] <-
+          c(blastOutput$q.start[entry]:blastOutput$q.end[entry])
+        finalOutputRangePointer <- finalOutputRangePointer + 1
+      }
+    }
+  }
+
+  #no matches were found, print how many were dropped
+  if(nrow(finalOutput) == 0){
+    if(!silent) {
+      message(sprintf("all %d matches were dropped, returning NULL", nrow(blastOutput)))
+    }
+    return(NULL)
+  }
+
+  #For the recursion, we need to store the start and end positions of our new
+  #input sequence with reference to the original sequence.  However, the recursion
+  #will just think it's a new sequence starting from index 1 to its length
+  #so after we blast the subsequence, we need to adjust the qstart and
+  #qend positions by the numbers we stored before the recursion.
+
+  finalOutput$q.start <- finalOutput$q.start + startPos - 1
+  finalOutput$q.end <- finalOutput$q.end + startPos - 1
+
+  #more spooky edge cases
+  minSide <- 0
+  maxSide <- 0
+
+  for(x in 1:length(finalOutputRange)){
+    finalOutputRange[[x]] <- finalOutputRange[[x]] + startPos - 1
+    if(minSide == 0 || min(finalOutputRange[[x]] < minSide)){
+      minSide <- min(finalOutputRange[[x]])
+    }
+
+    if(maxSide == 0 || max(finalOutputRange[[x]] > maxSide)){
+      maxSide <- max(finalOutputRange[[x]])
+    }
+
+  }
+
+
+  #Recursive step: if there are subsequences remaining after our right-most query that
+  #are >50aa in length, blast these as well
+
+  #added this because of random spooky edge cases
```

Comment:
The left most index was not properly being checked for overlap, and this lead to cases where multiple segments were being rerun continuously.